packages/css/src/plugins/merge-duplicate-at-rules.tsx

```diff
@@ -18,33 +17,38 @@ import type { ChildNode, AtRule } from 'postcss';
  * @media (min-width:500px){._171dak0l{border:2px solid red}._1swkri7e:before{content:'large screen'}}
  * ```
  */
-export const mergeDuplicateAtRules = plugin('merge-duplicate-at-rules', () => {
-  return (root) => {
-    const atRuleStore: Record<string, { node: AtRule; children: Record<string, ChildNode> }> = {};
+export const mergeDuplicateAtRules = (): Plugin => {
+  return {
+    postcssPlugin: 'merge-duplicate-at-rules',
+    Once(root) {
+      const atRuleStore: Record<string, { node: AtRule; children: Record<string, ChildNode> }> = {};
 
-    root.walkAtRules((atRule) => {
-      const name = atRule.name + atRule.params;
-      if (!atRuleStore[name]) {
-        atRuleStore[name] = {
-          node: atRule,
-          children: {},
-        };
-      }
-
-      atRule.each((node) => {
-        const stringifiedNode = node.toString();
-        if (!atRuleStore[name].children[stringifiedNode]) {
-          atRuleStore[name].children[stringifiedNode] = node;
+      root.walkAtRules((atRule) => {
```

Comment:
`walkAtRules` can be replaced with a top level `AtRule` instead of `Once`, and I believe `root` can instead have `atRule.root()`

Reply:
Due to the fact the atRule gets removed and later appended to root, I've split the work into an AtRule call and a OnceExit call that should run at the end. The tests stay valid with this approach.